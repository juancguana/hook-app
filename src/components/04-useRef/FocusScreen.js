import React, { useRef } from 'react';
import './effects.css';

export const FocusScreen = () => {
  const inputRef = useRef();

  const handleClick = () => {
    inputRef.current.select();
    console.log(inputRef);
  };

  return (
    <div>
      <h1>Focus Screen</h1>
      <hr />
      <input ref={inputRef} className='form-control' placeholder='Your Name' />
      <button className='btn btn-outline-primary mt-5' onClick={handleClick}>
        Focus
      </button>
    </div>
  );
};
