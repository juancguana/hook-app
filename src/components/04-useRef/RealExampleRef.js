import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';
import './effects.css';

export const RealExampleRef = () => {
  const [show, setShow] = useState(false);

  return (
    <div>
      <h1>Real example ref</h1>
      <hr />
      {show && <MultipleCustomHooks />}
      <br />
      <button className='btn btn-primary' onClick={() => setShow(!show)}>
        Show/Hide
      </button>
    </div>
  );
};
