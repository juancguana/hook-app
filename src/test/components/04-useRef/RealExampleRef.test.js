import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Test in <RealExample />', () => {
  const wrapper = shallow(<RealExampleRef />);

  test('Should show correctly', () => {
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });

  test('Should show the component', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  });
});
