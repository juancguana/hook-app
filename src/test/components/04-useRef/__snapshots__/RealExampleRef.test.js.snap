// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <RealExample /> Should show correctly 1`] = `
<div>
  <h1>
    Real example ref
  </h1>
  <hr />
  <br />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
